File: Day2/724-Find-Pivot-Index/Method2.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        int[] sum = new int[n];
        int leftSum = 0;

        for (int i = n - 1; i >= 0; i--) {
            if (i == n - 1) {
                sum[i] = nums[i];
                continue;
            }
            sum[i] = sum[i + 1] + nums[i];
        }

        for (int i = 0; i < n; i++) {
            leftSum += nums[i];
            if (leftSum == sum[i])
                return i;
        }

        return -1;
    }
}

File: Day3/169-Majority-Element/Method2.cpp
class Solution
{
public
    int majorityElement(int[] nums)
    {
        int count = 0, element = 0, n = nums.size();

        for (int num : nums)
        {
            if (count == 0)
            {
                element = num;
                count = 1;
            }
            else if (count > 0 && element == num)
                count++;
            else
                count--;
        }

        int freq = 0;
        for (int num : nums)
        {
            if (element == num)
                freq++;
        }

        return (freq > (n / 2)) ? element : 0;
    }
}